### PR TITLE
Multiple bug fixes

### LINKED ISSUE #1213: EELSSpectrum.estimate_elastic_scattering_intensity not working as expected
As `estimate_elastic_scattering_intensity` is used by default by `fourier_ratio_deconvolution`, the latter doesn't work as expected either.

Example:

```python

>>> from hyperspy.tests.signal.test_eels import Test_Estimate_Elastic_Scattering_Threshold
>>> t  = Test_Estimate_Elastic_Scattering_Threshold()
>>> t.setUp()
>>> t.signal.estimate_elastic_scattering_intensity(t.signal.estimate_elastic_scattering_threshold()).data
array([[ 0.,  0.],
       [ 0.,  0.],
       [ 0.,  0.]])
```

### PR DESCRIPTION
**Pull request by francisco-dlp**
#1213, #1212, #1211, #1210

### CODE REVIEW

File: hyperspy/signal.py [hunk 1]
```diff
@@ -2576,18 +2576,23 @@ def _iterate_signal(self):
...
         data = self.data.reshape(new_shape)
+        getitem = [0] * len(data.shape)
+        for axis in axes:
+            getitem[axis] = slice(None)
         for i in range(data.shape[unfolded_axis]):
-            getitem = [0] * len(data.shape)
-            for axis in axes:
-                getitem[axis] = slice(None)
             getitem[unfolded_axis] = i
             yield(data[getitem])
```
Comment (to266) [inline hyperspy/signal.py]: `tuple(getitem)`

Comment (francisco-dlp) [inline hyperspy/signal.py]: Would it make any difference? (Computing the tuple probably takes as much time or more than the performance improvement of using tuple vs list)

Comment (to266) [inline hyperspy/signal.py]: it's not for performance - slicing with a list vs slicing with a tuple are different things entirely!

Comment (francisco-dlp) [inline hyperspy/signal.py]: I agree, but not when the list always contains a slice, then they're equivalent. I'll change it though if that makes the code clearer.

**Comment by to266** [comment]
One of the new tests does not pass

**Comment by francisco-dlp** [comment]
It was due to a warning. I've moved it to the logger and now it should pass.

**Comment by to266** [comment]
it's due to the value being different than expected, and still fails....

**Comment by to266** [comment]
Still bad :(

**Comment by francisco-dlp** [comment]
I think that this is finally fine.

**Pull request merged**